### PR TITLE
Implemented the delete module page.

### PR DESCRIPTION
**Pull request by JoshRenton**
Had to change some things from the old classes and tests to accommodate for how I implemented it. There are 2 new classes, one for the page where you can choose to delete or add a module and the delete module class itself. I have also added tests for both classes.

### CODE REVIEW

**Comment by bridges-wood** [comment]
Closes #18

File: src/main/java/tempest/GUI/DeleteModulePage.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        return "deleteModulePage";
+    }
+
+    public JPanel getPanel(){
+        deleteModulePanel = new JPanel();
+        buttonPanel = new JPanel();
+        dropDownPanel = new JPanel();
+
+        cancelButton = new JButton("Cancel");
+        deleteButton = new JButton("Delete module");
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/DeleteModulePage.java:39]: You can use the Back Button class to do all the cancel button stuff for you

File: src/main/java/tempest/GUI/DeleteModulePage.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        deleteModulePanel.add(dropDownPanel);
+        deleteModulePanel.add(buttonPanel);
+
+        deleteModulePanel.setLayout(new BoxLayout(deleteModulePanel, BoxLayout.Y_AXIS));
+
+        return deleteModulePanel;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e){
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/DeleteModulePage.java:60]: Also we're tending to use spaces after `{` for methods and loops etc.

File: src/main/java/tempest/GUI/ManageModulesPage.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    public JPanel getPanel(){
+        manageModulesPanel = new JPanel();
+        pageSwapPanel = new JPanel();
+        cancelPanel = new JPanel();
+
+        dropDown = moduleDropDown.getModuleDropDown();
+
+        addModuleButton = new JButton("Add a module");
+        deleteModuleButton = new JButton("Delete a module");
+        cancelButton = new JButton("Cancel");
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/ManageModulesPage.java:37]: Same thing with the back button class here

File: src/main/java/tempest/GUI/ManageModulesPage.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        deleteModuleButton = new JButton("Delete a module");
+        backButton = new BackButton(manager);
+
+
+        addModuleButton.addActionListener(this);
+        deleteModuleButton.addActionListener(this);
+        backButton.addActionListener(this);
+
+        pageSwapPanel.add(addModuleButton);
+        pageSwapPanel.add(deleteModuleButton);
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/ManageModulesPage.java:46]: This is stupidly picky, but... atm we have the 'positive' button on the right (e.g. "Enter"), whilst the 'negative' button is on the left (e.g. "Back")

So if you don't mind could you just swap the order these two added to the panel

**Review by bridges-wood — APPROVED** [review]
Nice lambdas by the way.

**Comment by Drayz64** [comment]
> Nice lambdas by the way.

IntelliJ > VSCode

**Comment by bridges-wood** [comment]
> > Nice lambdas by the way.
> 
> IntelliJ > VSCode

Get this cringe off the repo. 😞

File: src/main/java/tempest/GUI/ManageModulesPage.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    public JPanel getPanel() {
+        manageModulesPanel = new JPanel();
+        pageSwapPanel = new JPanel();
+        backPanel = new JPanel();
+
+        dropDown = moduleDropDown.getModuleDropDown();
+
+        addModuleButton = new JButton("Add a module");
+        deleteModuleButton = new JButton("Delete a module");
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/ManageModulesPage.java:43]: If you look at the `homePage` you'll see we are now using the `LinkButton `component for buttons taking the user to other pages.

File: src/main/java/tempest/GUI/DeleteModulePage.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+        if (source == deleteButton) {
+            handleDeletingSession();
+        }
+    }
+
+    /**
+     * Handles deleting the session and updating the module drop down
+     */
+    private void handleDeletingSession() {
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/DeleteModulePage.java:76]: Stay away from my sessions, your job is to delete modules!

File: src/main/java/tempest/GUI/DeleteModulePage.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        // Checks if there are no modules to delete
+        if (dropDown.getItemCount() == 0) {
+            return;
+        }
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/DeleteModulePage.java:80]: I don't think this check is needed with the whole greying out the button thing, 

**it's up to you though** - keeping it means in the future we are able to scrap the greying out thing and not have to rewrite this check.

Comment (bridges-wood) [reply src/main/java/tempest/GUI/DeleteModulePage.java:80]: I think this is a good feature - better UX.

**Comment by bridges-wood** [comment]
Also when you're done, can you please tick off the tasks on the [issue](https://github.com/CM10313-Group-1/tempest/issues/18)? Thanks

File: src/main/java/tempest/GUI/DeleteModulePage.java (hunk 6 above)
Comment (JoshRenton) [reply src/main/java/tempest/GUI/DeleteModulePage.java:76]: I just realised that's named wrong. I will change it : )

File: src/main/java/tempest/GUI/DeleteModulePage.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import tempest.GUI.components.ModuleDropDown;
+
+public class DeleteModulePage extends Page {
+    private final State state;
+    private final GUIManager manager;
+    private final ModuleDropDown moduleDropDown = new ModuleDropDown();
+
+    private JPanel deleteModulePanel;
+    private JPanel buttonPanel;
+    private JPanel dropDownPanel;
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/DeleteModulePage.java:22]: IntelliJ is shouting at me that these 3 can be locals. I doubt we need to get these panels so no reason not to have them be local.

File: src/main/java/tempest/GUI/ManageModulesPage.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+import tempest.GUI.components.BackButton;
+import tempest.GUI.components.ModuleDropDown;
+
+public class ManageModulesPage extends Page {
+    private final GUIManager manager;
+    private final ModuleDropDown moduleDropDown = new ModuleDropDown();
+
+    private JPanel manageModulesPanel;
+    private JPanel pageSwapPanel;
+    private JPanel backPanel;
```
Comment (Drayz64) [inline src/main/java/tempest/GUI/ManageModulesPage.java:19]: Same here

File: src/main/java/tempest/GUI/DeleteModulePage.java (hunk 7 above)
Comment (JoshRenton) [reply src/main/java/tempest/GUI/DeleteModulePage.java:80]: The link buttons have messed up my tests now.

Comment (Drayz64) [reply src/main/java/tempest/GUI/DeleteModulePage.java:80]: How come? Your getters for the buttons still work right?

**Comment by JoshRenton** [comment]
All tests now pass again.

File: src/main/java/tempest/GUI/ManageModulesPage.java (hunk 9 above)
Comment (JoshRenton) [reply src/main/java/tempest/GUI/ManageModulesPage.java]: I have changed this now.

**Review by Drayz64 — APPROVED** [review]
I'm happy :)

**Comment by bridges-wood** [comment]
Good work, fun times. 🎉

**Pull request merged**